Makefile.include

```diff
@@ -318,23 +323,38 @@ LINKFLAGPREFIX ?= -Wl,
 
 DIRS += $(EXTERNAL_MODULE_DIRS)
 
-_LINK = $(if $(CPPMIX),$(LINKXX),$(LINK)) $(UNDEF) $(LINKFLAGPREFIX)--start-group $(BASELIBS) -lm $(LINKFLAGPREFIX)--end-group  $(LINKFLAGPREFIX)-Map=$(BINDIR)/$(APPLICATION).map $(LINKFLAGS)
+# Linker rule
+$(ELFFILE): FORCE  # Force rebuilding in case _LINK changes
```

Comment:
I find the comment confusing. Why should the variable change?

Reply:
You are right it's not clear.

I tried to prevent adding a new case where `make -B` would be required to rebuild if you changed LINKFLAGS for any reason.
For example in native by calling first `make all` then `make all-asan` or as developer by changing LINKFLAGS in one of the cpu/board makefile.include.

In the current state, during every build the elf file is re-linked even if nothing changed in the archives.
That is what I tried to mimic by putting the `FORCE` dependency here.

I welcome any suggestion for the comment :)